### PR TITLE
[WIP] USHIFT-1804: enable offline test instances

### PR DESCRIPTION
**Pull request by copejon**
This PR supports work on [USHIFT-1628](https://issues.redhat.com//browse/USHIFT-1628).  Offline testing requires an installer image with containers and ostree commits already baked in.  The current approach of publishing commits via network socket isn't feasible for offline testing.  Changes here address this with a new file ext type (`.edge-installer`), and the logic to handle it.

To build edge-installer images, composer-cli expects a) that the commit ref be reachable in the ostree repo and b) that an empty blueprint be pushed to composer-cli.  Once that's done, the image is built with composer-cli by specifying the empty blueprint name, the commit ref, the URL of the repo, and the "edge-installer" type. 

Enabling this build type requires a new mapping function for specifying filenames, blueprint contents, and embedded commit refs:

**Filenames**: must end with `.edge-installer`.  The file's name specifies the to-be-embedded commit-ref's blueprint filename.  E.g. `rhel92-source-isolated.edge-installer` is decomposed to blueprint-file `rhel92-source-isolated`, image type `edge-installer`.

**File contents**: must contain an empty blueprint (see ./test/image-blueprints/group3/rhel92-source-isolated.edge-installer).  This is used to specify the base image for the installer.  All packages and customization must be defined in the commit-ref's blueprint, NOT the empty blueprint.

TODO
- [ ] Update ./test/README.md

### CODE REVIEW

File: test/image-blueprints/group3/rhel92-source-isolated.edge-installer [hunk 1]
```diff
@@ -0,0 +1,6 @@
+name = "rhel-9.2-source-isolated-installer"
```
Comment (pmtk) [inline test/image-blueprints/group3/rhel92-source-isolated.edge-installer:1]: Should the file be named `rhel-9.2-microshift-source-isolated`?

```
ERROR: OSTreeOptionsError: ostree repository "http://10.192.10.219:8080/repo/refs/heads/rhel92-source-isolated" returned status: 404 Not Found
```

```
* rhel-9.2-microshift-source-isolated
    Latest Commit (17.2 kB):
      3ce75e1151536e0b9c34550d49124cb76abe234c9cb0508f97bee23404fb97ac
    Version (ostree.commit.version): 9.2
    Timestamp (ostree.commit.timestamp): 2023-10-25T22:47:23+00

```

File: test/kickstart-templates/kickstart-offline.ks.template [hunk 2]
```diff
@@ -0,0 +1,82 @@
+lang en_US.UTF-8
+keyboard us
+timezone UTC
+text
+reboot
+
+# Do not enable a network device
+network --no-activate --nodefroute
```
Comment (ggiguash) [inline test/kickstart-templates/kickstart-offline.ks.template:8]: We need to set a host name here

**Comment by copejon** [comment]
per our discussion, I'll close this PR and we can address whether to use edge-installer images in the future.

**Pull request closed without merging**